src/ngx_http_lua_args.c

```diff
@@ -544,7 +547,119 @@ ngx_http_lua_ffi_req_get_uri_args(ngx_http_request_t *r, u_char *buf,
 
     return i;
 }
-#endif /* NGX_LUA_NO_FFI_API */
 
 
+int
+ngx_http_lua_ffi_req_get_uri_args(ngx_http_request_t *r, u_char *buf,
+    ngx_http_lua_ffi_table_elt_t *out, int count)
+{
+    ngx_memcpy(buf, r->args.data, r->args.len);
+
+    return ngx_http_lua_ffi_req_args_helper(buf, r->args.len, out, count);
+}
+
+
+int
+ngx_http_lua_ffi_req_validate_post(ngx_http_request_t *r)
+{
+    if (r->connection->fd == (ngx_socket_t) -1) {
+        return NGX_HTTP_LUA_FFI_BAD_CONTEXT;
+    }
+
+    if (r->discard_body) {
+        return NGX_HTTP_LUA_FFI_POST_ARGS_EMPTY;
+    }
+
+    if (r->request_body == NULL) {
+        return NGX_HTTP_LUA_FFI_POST_ARGS_UNREAD;
+    }
+
+    if (r->request_body->temp_file) {
+        return NGX_HTTP_LUA_FFI_POST_ARGS_IN_FILE;
+    }
+
+    if (r->request_body->bufs == NULL) {
+        return NGX_HTTP_LUA_FFI_POST_ARGS_EMPTY;
+    }
+
+    return NGX_HTTP_LUA_FFI_POST_ARGS_OK;
```

Comment:
We usually do not use a separate C function to do checkups. See other existing FFI C functions for error handling.